docs/adr/0006-forms.md

```diff
@@ -0,0 +1,57 @@
+# Forms
+
+Date: 2020-09-16
+
+## Status
+
+2020-09-16 Proposed
+
+## Context
+
+Currently the form library [`react-reactive-forms`](https://github.com/bietkul/react-reactive-form) is being used. This poses the following problems:
+
+- It is difficult and time consuming to customize the form to specific needs and requirements, because the form logic consists of many layers and you have indirect control over the end result.
+  - Many forms are defined through a configuration.
+  - Forms are created with a form builder.
+  - Form elements are defined through components
+  - Rendering is defined throught a prop on a field component.
+- There is no easy interaction through hooks.
+- Not actively maintained anymore (latest release from 2018-08-19).
+
+It does provide us with the following benefits:
+
+- Additional questions in the incident form can come from the backend in a json definition and easily be inserted into the form.
+- Has built-in validation.
+- Small footprint, no dependencies.
+
+## Considerations
+
+We would like to update how we develop forms. The following requirements are
+important to us:
+
+- *Should have* low footprint, few dependencies.
+- *Should be* easy to customize, or define custom components, validation, localization.
+- *Should have* no re-rendering on changes to form fields
+- *Must have* integration with React hooks.
+- *Must have* validation.
+
+
+| Package                                                               | No run-time deps. | < 10 kB | Hooks | Validation | Customizable | No re-render | Stargazers | Updated        |
+| --------------------------------------------------------------------- | ---------------- | -------- | ----- | ---------- | ------------ | ------------ | ---------- | -------------- |
+| [formik](https://github.com/jaredpalmer/formik)                       | ❌                | ❌        | ❌     | ✅*         | ✅            | ❌            | 24.5k      | 2 months ago   |
+| [react-hook-form](https://github.com/react-hook-form/react-hook-form) | ✅                | ✅        | ✅     | ✅**        | ✅            | ✅            | 14.1k      | recently       |
+| plain HTML                                                            | ✅                | ✅        | ❌     | ✅          | ✅            | ✅            | –          | –              |
+| [react-reactive-form](https://github.com/bietkul/react-reactive-form) | ✅                | ✅        | ❌     | ✅          | ✅            | ❌            | 250        | 25 months ago  |
```

Comment:
What I'm missing here is the option to use our own (controlled) components. Since we're dealing with components coming from `@datapunt/asc-ui`, this is a hard requirement.

Reply:
The column `customizable` refers to the bullet "*Should be* easy to customize, or use custom components, validation, localization." So, using custom components is covered in that column.

Reply:
Formik definitely doesn't support custom components, so I would split that column up in its appropriate parts, because now we're losing overview because of grouping of criteria.